widgetssdk/src/main/java/com/glia/widgets/chat/ChatActivity.kt

```diff
@@ -95,14 +60,11 @@ class ChatActivity : FadeTransitionActivity() {
         chatView.setOnEndListener(::finishAndRemoveTask)
 
         chatView.setOnMinimizeListener(::finish)
-        chatView.setOnNavigateToCallListener(this::startCallScreen)
-        chatView.startChat(
-            engagementConfiguration.companyName,
-            engagementConfiguration.queueIds,
-            engagementConfiguration.contextAssetId,
-            engagementConfiguration.screenSharingMode,
-            engagementConfiguration.chatType ?: ChatType.LIVE_CHAT
-        )
+
+        val chatType = intent.getIntExtra(ExtraKeys.CHAT_TYPE, -1).takeIf { it != -1 }
+            ?.let { ChatType.values()[it] } ?: ChatType.LIVE_CHAT
```

Comment:
Could you extract this code to some to some `IntentExtentions` class with a method name `fun Intent.getChatType` and also extract any calls to `intent.putExtra(CHAT_TYPE, ChatType.SECURE_MESSAGING as Parcelable)` to the same class with a name `intent.putChatType()`. 

I know that technically your solution is completely safe even for any future changes to enums but I really would still prefer to have such logic in one place.